.pre-commit-hooks.yaml

```diff
@@ -0,0 +1,8 @@
+- id: regal-lint
+  stages:
+    - pre-commit
+  language: golang
```

Comment:
This should be `system`, IIRC. We're calling the `regal` binary here, not depending on the Golang tooling.

Reply:
I've went with `golang` as that ensures `pre-commit` installs the `regal` binary for the given tag in this repository, thus maintenance of the hook is simplified if anything needs to change / be added.

If we use `system` it requires the user has a pre-existing installation of `regal` that they'd need to keep updated.

Reply:
I suppose we could have to options, one which was `regal-lint-go` (or whatever) and one `regal-lint` or something like that.

Reply:
I'd be in favor of `system` - the casual OPA user or policy editor might not even have Golang installed.

Reply:
probably a wrapper script to install the right binary for the current architecture might be handy though.

Reply:
To summarise:

1) `language: golang`

`pre-commit` will install the tagged version by doing a `go install ./...` in an isolated `GOPATH`.

In this scenario the consumer doesn't have to have `regal` on their `PATH`, but needs the `go` build chain.  `pre-commit` will use the tagged version of `regal` from the repo.  If `regal` does exist on `PATH` the user _may_ experience differences in functionality if the versions are misaligned.

2) `language: system`

`pre-commit` will not install `regal` and will simply execute whatever version is on `PATH`.

In this scenario the consumer doesn't have to have `go` build chain on their `PATH`, but needs the `regal` binary.  `pre-commit` will fail if `regal` does not exist.  There is an additional risk, likely small (?), that if the _interface_ of the `regal` binary changed the `pre-commit` hook would fail due to version mismatching.

Happy to take your guidance here; personally I favour (1).

Reply:
I could add hooks with `id: regal-lint` and `id: regal-lint-from-system` to give both options?

Reply:
That sounds like a good middle ground to me 👍

Reply:
Updated 👍

Reply:
Thanks! Do you think you could add a section on this to the README? I'm thinking a new one called "Pre-Commit Hooks" below the "Inline Ignore Directives" section, where you provide instructions for how to install the two different versions, and how they're different?

If not, I can have it added later.

Reply:
Updated with README as request